src/flows/zome-call.ts

```diff
@@ -44,11 +58,23 @@ export default (publicClient, internalClient) => async (call: CallRequest) => {
     signature = 'TODO-look-into-hClient-signature'
   }
 
-  const requestData = buildServiceLoggerRequestPackage(call)
+  const dna = await lookupDnaByHandle(masterClient, happId, handle)
+  const dnaHash = dna.hash
+  // see if this instance is actually hosted, we may have to get the host's instance if not
+  const instance = await lookupHoloInstance(publicClient, {agentId, dnaHash})
+  // use the looked-up instance info, not the info passed in to the zome call
+  const instanceId = instanceIdFromAgentAndDna(instance)
+
+  const requestData = buildServiceLoggerRequestPackage({
+    dnaHash,
+    zome: zomeName,
```

Comment:
If other functions use variable `zomeName`, perhaps we could rename this here, and hide this renaming into the function itself?